src/api/Controllers/WorkflowController.cs

```diff
@@ -12,28 +13,102 @@ public class WorkflowController : ControllerBase
+    /// </summary>
+    [HttpPost("change")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    [SwaggerResponse(StatusCodes.Status200OK, "Applies a workflow change and updates the workflow.")]
+    [SwaggerResponse(StatusCodes.Status400BadRequest, "The workflow could not be created due to invalid input.")]
+    [SwaggerResponse(StatusCodes.Status401Unauthorized, "The current user is not authorized to create workflow changes.")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "The server encountered an unexpected condition that prevented it from fulfilling the request.")]
+    public async Task<IActionResult> ApplyWorkflowChangeAsync([FromBody] WorkflowChangeRequest workflowChangeRequest)
+    {
+        if (!ModelState.IsValid) return BadRequest(ModelState);
+
+        var subjectId = HttpContext.GetUserSubjectId();
+
+        // Check permission for editing the borehole
+        if (!await boreholePermissionService.CanEditBoreholeAsync(subjectId, workflowChangeRequest.BoreholeId, true).ConfigureAwait(false)) return Unauthorized();
+
+        // Check permissions specifically for publishing the borehole
+        if (workflowChangeRequest.NewStatus == WorkflowStatus.Published &&
+            !await boreholePermissionService.HasUserRoleOnWorkgroupAsync(subjectId, workflowChangeRequest.BoreholeId, Role.Publisher).ConfigureAwait(false))
+        {
+            return Unauthorized();
+        }
+
+        var workflow = await context.WorkflowsV2WithIncludes.SingleOrDefaultAsync(w => w.BoreholeId == workflowChangeRequest.BoreholeId).ConfigureAwait(false);
         if (workflow == null)
         {
-            return NotFound();
+            var workflowNotFoundMessage = $"Workflow for borehole with {workflowChangeRequest.BoreholeId} not found.";
+            logger?.LogWarning(workflowNotFoundMessage);
+            return NotFound(workflowNotFoundMessage);
         }
 
-        return Ok(workflow);
+        var newAssignee = await context.Users.AsNoTracking().SingleOrDefaultAsync(u => u.Id == workflowChangeRequest.NewAssigneeId).ConfigureAwait(false);
+        if (newAssignee == null)
+        {
+            var userNotFoundMessage = $"New assignee with id {workflowChangeRequest.NewAssigneeId} not found.";
+            logger?.LogWarning(userNotFoundMessage);
+            return NotFound(userNotFoundMessage);
+        }
+
+        var user = await context.Users.AsNoTracking().SingleOrDefaultAsync(u => u.SubjectId == subjectId).ConfigureAwait(false);
+
+        // Create the change history entry
+        var change = new WorkflowChange
+        {
+            WorkflowId = workflow.Id,
+            FromStatus = workflow.Status,
+            ToStatus = workflowChangeRequest.NewStatus ?? workflow.Status,
+            Comment = workflowChangeRequest.Comment ?? "",
+            Created = DateTime.UtcNow,
+            CreatedById = user.Id,
+            AssigneeId = workflowChangeRequest.NewAssigneeId,
```

Comment:
PP: `newAssignee.Id` anstatt die id aus dem workflowChangeRequest.
```suggestion
            AssigneeId = newAssignee.Id,
```